docs/docs/files-gatsby-looks-for-in-a-plugin.md

```diff
@@ -1,10 +1,23 @@
 ---
-title: Files Gatsby Looks for in a Plugin
+title: Creating a Generic plugin
 ---
 
+This section is aimed at explaining the structure of a gatsby plugin and the files that are needed to create it.
+
+As seen in the [what is a plugin doc](/docs/what-is-a-plugin/) is a piece of software that acts as an add-on and gives your site additional functionality.
+
+They contain a file, usually in the project root, called `package.json` - this file holds various metadata relevant to the project. This file is used to give information to npm that allows it to identify the project as well as handle the project's dependencies.
+The process of initializing a `package.json` in a project is done by running :
+
+```shell
+  npm init
+```
+
 ## What files does Gatsby look for in a plugin?
 
-All files are optional unless specifically marked as required.
+After running the initialization command. You would need to provide information for the `package.json`
```

Comment:
Since `npm init` lets you add in information for a bunch of these fields maybe this should say something like this:

```suggestion
You can provide more information for fields in the `package.json` that weren't prompted by running the initialization command.
```